src/compton/test/tCompton_NWA.cc

```diff
@@ -0,0 +1,244 @@
+//---------------------------------------------------------------------------//
+// $Id$
+//---------------------------------------------------------------------------//
+
+#include "compton/Compton_NWA.hh"
+#include "ds++/Release.hh"
+#include "ds++/SP.hh"
+#include "ds++/ScalarUnitTest.hh"
+#include "ds++/Soft_Equivalence.hh"
+
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+
+namespace rtt_compton_test {
+
+using rtt_dsxx::SP;
+using rtt_dsxx::soft_equiv;
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+//!  Tests the Compton_NWA constructor and a couple of access routines.
+void compton_file_test(rtt_dsxx::UnitTest &ut) {
+  // Start the test.
+
+  std::cout << "\n------------------------------------------" << std::endl;
+  std::cout << "   Test Draco code calling NWA routines" << std::endl;
+  std::cout << "------------------------------------------" << std::endl;
+
+  // open a small mg opacity file:
+  std::string filename = "mg_ascii.compton";
+  std::cout << "Attempting to construct a Compton_NWA object...\n" << std::endl;
+  std::shared_ptr<rtt_compton::Compton_NWA> compton_test;
+
+  try {
+    compton_test.reset(new rtt_compton::Compton_NWA(filename));
+  } catch (int asrt) {
+    FAILMSG("Failed to construct a Compton_NWA object!");
+    // if construction fails, there is no reason to continue testing...
+    return;
+  }
+  std::cout << "\n(...Success!)" << std::endl;
+
+  // Check some of the data in the NWA-opened file:
+  std::vector<double> grp_bds = compton_test->get_group_bounds();
+  std::vector<double> etemp_evals = compton_test->get_etemp_pts();
+
+  Ensure(grp_bds.size() == 2);
+  Ensure(etemp_evals.size() == 7);
+
+  if (!soft_equiv(grp_bds[0], 3.91389000e-02)) {
+    std::cout << "Lower group bound read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(grp_bds[1], 5.87084000e-02)) {
+    std::cout << "Upper group bound read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (!soft_equiv(etemp_evals[0], 1.76377944e-05)) {
+    std::cout << "Etemp eval read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(etemp_evals[6], 6.75507064e-04)) {
+    std::cout << "Etemp eval read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (!soft_equiv(compton_test->get_min_etemp(), 1e-6)) {
+    std::cout << "Min etemp read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+  if (!soft_equiv(compton_test->get_max_etemp(), 7e-4)) {
+    std::cout << "Max etemp read incorrectly!" << std::endl;
+    ITFAILS;
+  }
+
+  if (ut.numFails == 0) {
+    std::cout << "\nCorrectly read group bounds and electron temps!"
+              << std::endl;
+  }
+
+  // try "interpolating" at one of the exact eval points in the test library,
+  // and check the result:
+  std::vector<std::vector<std::vector<double>>> interp_data =
+      compton_test->interpolate(4.87227167e-04);
```

Comment:
Create a variable to hold 4.8722... that has a useful name so that this test is easier for a novice to understand.